Do not use the C compiler for compiling regexes at Maxima startup.
This reverts Maxima commit 4226ee2.  Introduced in Trac #17375
(Maxima should not need a working C compiler to run).

--- a/src/commac.lisp
+++ b/src/commac.lisp
@@ -376,10 +376,10 @@
       (or (strip-float-zeros string) string)
       string)))
 
-(defparameter trailing-zeros-regex-f-0 (compile nil (maxima-nregex::regex-compile "^(.*\\.[0-9]*[1-9])00*$")))
-(defparameter trailing-zeros-regex-f-1 (compile nil (maxima-nregex::regex-compile "^(.*\\.0)00*$")))
-(defparameter trailing-zeros-regex-e-0 (compile nil (maxima-nregex::regex-compile "^(.*\\.[0-9]*[1-9])00*([^0-9][+-][0-9]*)$")))
-(defparameter trailing-zeros-regex-e-1 (compile nil (maxima-nregex::regex-compile "^(.*\\.0)00*([^0-9][+-][0-9]*)$")))
+(defvar trailing-zeros-regex-f-0 (coerce (maxima-nregex::regex-compile "^(.*\\.[0-9]*[1-9])00*$") 'function))
+(defvar trailing-zeros-regex-f-1 (coerce (maxima-nregex::regex-compile "^(.*\\.0)00*$") 'function))
+(defvar trailing-zeros-regex-e-0 (coerce (maxima-nregex::regex-compile "^(.*\\.[0-9]*[1-9])00*([^0-9][+-][0-9]*)$") 'function))
+(defvar trailing-zeros-regex-e-1 (coerce (maxima-nregex::regex-compile "^(.*\\.0)00*([^0-9][+-][0-9]*)$") 'function))
 
 ;; Return S with trailing zero digits stripped off, or NIL if there are none.
 
